OneSignalSDK/onesignal/core/src/main/java/com/onesignal/notifications/INotificationLifecycleListener.kt

```diff
@@ -0,0 +1,29 @@
+package com.onesignal.notifications
+
+/**
+ * Implement this interface and provide an instance to [INotificationsManager.addForegroundLifecycleListener]
+ * in order to receive control during the lifecycle of the notification. This handler will *only* receive
+ * control when the application is in the foreground when the notification is received.
+ *
+ * @see [Foreground Notification Received Event | OneSignal Docs](https://documentation.onesignal.com/docs/sdk-notification-event-handlers#foreground-notification-received-event)
+ */
+interface INotificationLifecycleListener {
+
+    /**
+     * Called when a notification is to be displayed to the user. This callback
+     * gives the implementor the ability to prevent the notification from displaying to the
+     * user entirely.
+     *
+     * *Note:* this runs after the Notification Service Extension [INotificationServiceExtension]
+     * has been called (if one exists), which has the following differences:
```

Comment:
> Note:* this runs after the Notification Service Extension [INotificationServiceExtension]
    has been called (if one exists), which has the following differences:

Revise the wording for clarity 👇🏽 

> If a Notification Service Extension [INotificationServiceExtension] exists, `onWillDisplay` will run after the extension is called...